apps/07jquery-tic-tac-toe/script.js

```diff
@@ -1,5 +1,65 @@
 'use strict';
-
+//$(document).ready(function(){})
 $(document).on('ready', function() {
-  // Put app logic in here
+    // Put app logic in here
+  var playerTurn = 'X';
+  function checkForWin(){
+      // console.log('yay!');
+      // debugger;
+    if (
+          // horizontal rows
+            (($('[data-cell=0]').text() === playerTurn) &&
+            ($('[data-cell=1]').text() === playerTurn) &&
+            ($('[data-cell=2]').text() === playerTurn)) ||
+            (($('[data-cell=3]').text() === playerTurn) &&
+            ($('[data-cell=4]').text() === playerTurn) &&
+            ($('[data-cell=5]').text() === playerTurn)) ||
+            (($('[data-cell=6]').text() === playerTurn) &&
+            ($('[data-cell=7]').text() === playerTurn) &&
+            ($('[data-cell=8]').text() === playerTurn)) ||
+            // vertical columns
+            (($('[data-cell=0]').text() === playerTurn) &&
+            ($('[data-cell=3]').text() === playerTurn) &&
+            ($('[data-cell=6]').text() === playerTurn)) ||
+            (($('[data-cell=1]').text() === playerTurn) &&
+            ($('[data-cell=4]').text() === playerTurn) &&
+            ($('[data-cell=7]').text() === playerTurn)) ||
+            (($('[data-cell=2]').text() === playerTurn) &&
+            ($('[data-cell=5]').text() === playerTurn) &&
+            ($('[data-cell=8]').text() === playerTurn)) ||
+            // diagonal lines
+            (($('[data-cell=0]').text() === playerTurn) &&
+            ($('[data-cell=4]').text() === playerTurn) &&
+            ($('[data-cell=8]').text() === playerTurn)) ||
+            (($('[data-cell=6]').text() === playerTurn) &&
+            ($('[data-cell=4]').text() === playerTurn) &&
+            ($('[data-cell=2]').text() === playerTurn))) {
+              // console.log('yay!');
+              // debugger;
+      return $('#announce-winner').text('Player ' + playerTurn + ' Wins!');
+    } else if (($('[data-cell=0]').text() !== '')&&($('[data-cell=1]').text() !== '')&&($('[data-cell=2]').text() !== '')&&($('[data-cell=3]').text() !== '')&&($('[data-cell=4]').text() !== '')&&($('[data-cell=5]').text() !== '')&&($('[data-cell=6]').text() !== '')&&($('[data-cell=6]').text() !==           '')&&($('[data-cell=8]').text() !== '')){
+      return $('#announce-winner').text('You Loose, You Loser!');
```

Comment:
Man, pretty savage return logic!

To shorten your code, you could also increment a moveCounter variable with every valid turn or make use of the .each() jQuery method.

ex:
```
var moveCounter = 0;
$('[data-cell]').each(function() {
     if($(this).text()) {
       moveCounter++
     }
});


... else if(moveCounter === 9)  { 
      return $('#announce-winner').text('You Lose, You Loser!');
}
```

`$(this).text()` will return false if the data-cell contains an empty string because `''` is a falsey value.